### PR TITLE
pkg: lwip: some fixes

### PR DESCRIPTION
**Pull request by miri64**
During hacking for #5802 I found some minor bugs in lwIP. This PR fixes those.

### CODE REVIEW

File: pkg/lwip/contrib/netdev2/lwip_netdev2.c [hunk 1]
```diff
@@ -47,7 +47,6 @@
 #define ETHERNET_IFNAME2 'T'
 
 /* running number for different interfaces */
-static uint8_t _num = 0;
```
Comment (miri64) [inline pkg/lwip/contrib/netdev2/lwip_netdev2.c]: I stated this already in the commit message, but something like this already happens lwIP internally so no need for this and potentially makes things just more confusing ;-)

**Comment by miri64** [comment]
Ping? This is a really low-hanging fruit ;-)

File: pkg/lwip/contrib/netdev2/lwip_netdev2.c [hunk 2]
```diff
@@ -47,7 +47,6 @@
 #define ETHERNET_IFNAME2 'T'
 
 /* running number for different interfaces */
```
Comment (kaspar030) [inline pkg/lwip/contrib/netdev2/lwip_netdev2.c]: also remove comment?

**Comment by miri64** [comment]
Done.

**Comment by aabadie** [comment]
Seems like everybody approves the changes here. Merging !

**Pull request merged**